internal/pkg/gateway/submit.go

```diff
@@ -88,26 +84,20 @@ loop:
 			return nil, newRpcError(codes.DeadlineExceeded, "submit timeout expired while broadcasting to ordering service")
 		}
 	}
-	logger.Warnw("Insufficient number of orderers could successfully process transaction to satisfy quorum requirement", "successes", successes, "quorum", quorum)
-	return nil, newRpcError(codes.Unavailable, "insufficient number of orderers could successfully process transaction to satisfy quorum requirement", errDetails...)
+	if successes == 0 {
```

Comment:
if we submitted to less than a quorum, inclusion of our transaction is not guaranteed.

Reply:
we're not waiting for the orderers to give us a result
we expect results from peers

Reply:
No, in the gateway we wait for results from the orderers because we want to be certain we submitted to enough orderers so the transaction will be included even if the orderers are malicious.

Reply:
In theory we should expect a result, but we don't get a result here. The result comes at the peers and is checked there as well

Reply:
In the version that is now in main, I have no confidence in the error message. The error message will often appear falsely.

The client logic is as follows: I perform an operation, if the operation was not successful, then it is definitely unsuccessful. In your case, there is an error message, but the transaction was successful. False error return.

Reply:
The client listens to the transaction resultss through a different API, not though this one. 

This API (submit) is used to submit transactions to the orderer.

At the time of transaction submission you cannot know if the transaction will be in, or not. 

Therefore, it's classified as a failure because you are not 100% certain.

This is different from the semantics of Raft, where sending to the orderer may result in the transaction not being in the block.

Reply:
I'm sorry, but the client's pattern is always the same:
.........
- sent to the orderer(s)
- waiting for the result from the peers
- what happened? why is my transaction gone? are you sure it's gone? maybe I just timed out.
..........

Reply:
The Fabric core dictates how clients should act, and not the other way around. 

If sending to the orderers fails, the client should retry, but in any case it should be aware that the transaction might be included in a block in the future, even if the gateway returned a failure, and the application (client side and chaincode) should rely on persistent finalized data (blockchain, world state).

Reply:
Here's your [test](https://github.com/hyperledger/fabric/pull/4414). I took out Resubmitting. I didn't add anything, just removed it. And it works. Run it 1,000 times, it will execute 1,000 times.

Reply:
So Resubmitting is not necessary. Is there a test that would show that Resubmitting is necessary?

Reply:
You removed a line from a test and it still succeed. I don't understand how this means you need to change the production code?

Reply:
I removed the code that in the test showed the need for what you require from the client: Resubmitting the same transaction.

There is no test that shows the need for Resubmitting

Reply:
Is there a test that would show the need for Resubmitting?

Sorry, but so far the current implementation shows unnecessary error and unnecessary Resubmitting.
This is what compelled me to start the corresponding discussion.

Reply:
Why do you think there is a *need* for resubmitting? The test resubmits, and checks it doesn't fail. It does not mean you need to resubmit.

Reply:
That's not my opinion. It is the test that thinks so. 
And as soon as I remove this step, and the test is executed successfully and will always be executed, then the repeated resubmitting and returning of the error becomes meaningless. Because the error didn't happen.

Reply:
A test might test a specific step does not fail, but it does not mean that this step is essential. It just tests that it does not fail if you do that. It's not the same as testing that it is required.

Reply:
By the way, just a reminder that this is a continuation of a past [discussion](https://github.com/hyperledger/fabric/pull/3975#discussion_r1091543576). I still do not agree with this decision.

For me, this step was the only one that could show the advantage of your solution. 
It turns out that there is no test that would show the correctness of the existing solution.